plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/AppEngineProjectDeployer.java

```diff
@@ -0,0 +1,88 @@
+package com.google.cloud.tools.eclipse.appengine.deploy;
+
+import java.io.File;
+import java.util.Collections;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.core.runtime.SubMonitor;
+import org.w3c.dom.Document;
+import org.w3c.dom.NodeList;
+
+import com.google.cloud.tools.appengine.api.deploy.DefaultDeployConfiguration;
+import com.google.cloud.tools.appengine.cloudsdk.CloudSdk;
+import com.google.cloud.tools.appengine.cloudsdk.CloudSdkAppEngineDeployment;
+
+public class AppEngineProjectDeployer {
+
+  private static final String PLUGIN_ID = "com.google.cloud.tools.eclipse.appengine.localserver"; //$NON-NLS-1$
+  private static final String APPENGINE_WEB_XML_PATH = "WEB-INF/appengine-web.xml"; //$NON-NLS-1$
+
+  public void deploy(IPath stagingDir, CloudSdk cloudSdk, IProgressMonitor monitor) throws CoreException {
+    SubMonitor progress = SubMonitor.convert(monitor, 1);
+    progress.setTaskName(Messages.getString("task.name.deploy.project")); //$NON-NLS-1$
+    try  {
+
+      Document doc = parseAppEngineWebXml(new File(stagingDir.append(APPENGINE_WEB_XML_PATH).toOSString()));
+
+      String projectId = getProjectId(doc);
+      if (projectId == null) {
+        throw new CoreException(new Status(IStatus.ERROR, PLUGIN_ID, Messages.getString("project.id.missing"))); //$NON-NLS-1$
+      }
+
+      String projectVersion = getProjectVersion(doc);
+      if (projectVersion == null) {
+        throw new CoreException(new Status(IStatus.ERROR, PLUGIN_ID, Messages.getString("project.version.missing"))); //$NON-NLS-1$
+      }
+
+      DefaultDeployConfiguration deployConfig = new DefaultDeployConfiguration();
+      deployConfig.setDeployables(Collections.singletonList(stagingDir.append("app.yaml").toFile())); //$NON-NLS-1$
+      deployConfig.setProject(projectId);
+      deployConfig.setVersion(projectVersion);
+      deployConfig.setPromote(true);
```

Comment:
I agree with @briandealwis , it should be configurable and default to false. The deployed version's URL will be displayed in the console, so users can access it, but their prod version is still available on the default URL. Perhaps we can display the URL more prominently by decorating the console output to make it a link. I'll file an issue for both the configurability and the console output enhancement.
And make it false by default.